File: opensource/kernel/src/main/java/com/sos/scheduler/engine/kernel/variable/UnmodifiableVariableSet.java
package com.sos.scheduler.engine.kernel.variable;

import com.google.common.collect.ImmutableCollection;

import javax.annotation.Nullable;

public interface UnmodifiableVariableSet {
    @Nullable String tryGet(String name);
    String get(String name);
    ImmutableCollection<String> getNames();
}

File: opensource/kernel/src/main/java/com/sos/scheduler/engine/kernel/variable/UnmodifiableVariableSet.scala
package com.sos.scheduler.engine.kernel.variable

import com.google.common.collect.ImmutableMap

trait UnmodifiableVariableSet {
  def apply(name: String): String

  def size: Int

  def keys: Iterable[String]

  def toMap: Map[String, String]

  /** Für Java. */
  def toGuavaMap: ImmutableMap[String, String]

  /** Für Java. */
  def getNames: java.util.Collection[String]
}